lms/static/js/spec/search/search_spec.js

```diff
@@ -285,145 +194,372 @@ define([
+            it('displays course name in breadcrumbs', function () {
+                expect(this.seqItem.$el.find('.sri-course-name')).toContainHtml(this.model.get('course_name'));
+            });
+            it('logs view event', logsSearchItemViewEvent);
         });
 
-        it('shows error message', function () {
-            this.listView.showErrorMessage();
-            expect($('#course-content')).toBeHidden();
-            expect(this.listView.$el).toBeVisible();
-            expect(this.listView.$el).not.toBeEmpty();
+    });
+
+
+    describe('SearchForm', function () {
+
+        function trimsInputString() {
+            var term = '    search string  ';
+            $('.search-field').val(term);
+            $('form').trigger('submit');
+            expect(this.onSearch).toHaveBeenCalledWith($.trim(term));
+        }
+
+        function doesSearch() {
+            var term = '  search string  ';
+            $('.search-field').val(term);
+            this.form.doSearch(term);
+            expect(this.onSearch).toHaveBeenCalledWith($.trim(term));
+            expect($('.search-field').val()).toEqual(term);
+            expect($('.search-field')).toHaveClass('is-active');
+            expect($('.search-button')).toBeHidden();
+            expect($('.cancel-button')).toBeVisible();
+        }
+
+        function triggersSearchEvent() {
+            var term = 'search string';
+            $('.search-field').val(term);
+            $('form').trigger('submit');
+            expect(this.onSearch).toHaveBeenCalledWith(term);
+        }
+
+        function changesToActiveState() {
+            var term = 'search string';
+            $('.search-field').val(term);
+            $('form').trigger('submit');
+            expect($('.search-field')).toHaveClass('is-active');
+            expect($('.search-button')).toBeHidden();
+            expect($('.cancel-button')).toBeVisible();
+        }
+
+        function clearsSearchOnCancel() {
+            $('.search-field').val('search string');
+            $('.cancel-button').trigger('click');
+            expect($('.search-field')).not.toHaveClass('is-active');
+            expect($('.search-button')).toBeVisible();
+            expect($('.cancel-button')).toBeHidden();
+            expect($('.search-field')).toHaveValue('');
+        }
+
+        function clearsSearchOnEmpty() {
+            $('.search-field').val('');
+            $('form').trigger('submit');
+            expect(this.onClear).toHaveBeenCalled();
+            expect($('.search-field')).not.toHaveClass('is-active');
+            expect($('.cancel-button')).toBeHidden();
+            expect($('.search-button')).toBeVisible();
+        }
+
+        describe('CourseSearchForm', function () {
+            beforeEach(function () {
+                loadFixtures('js/fixtures/search/course_search_form.html');
+                this.form = new CourseSearchForm();
+                this.onClear = jasmine.createSpy('onClear');
```

Comment:
Upon taking a better look you are right. You can't spyOn collection event handlers that are attached from you view init.